jwt-native/src/main/java/org/ballerinalang/stdlib/jwt/JwksClient.java

```diff
@@ -0,0 +1,162 @@
+                } catch (NoSuchAlgorithmException | KeyManagementException e) {
+                    return createError("Failed to init SSL context. " + e.getMessage());
+                }
+            }
+            BMap<BString, BString> trustStore =
+                    (BMap<BString, BString>) getMapValueIfPresent(secureSocket, Constants.TRUSTSTORE);
+            if (trustStore != null) {
+                try {
+                    SSLContext sslContext = initSslContext(trustStore);
+                    HttpClient client = buildHttpClient(httpVersion, sslContext);
+                    return callJwksEndpoint(client, url.getValue());
+                } catch (Exception e) {
+                    return createError("Failed to init SSL context with truststore. " + e.getMessage());
+                }
+            }
+        }
+        HttpClient client = buildHttpClient(httpVersion);
+        return callJwksEndpoint(client, url.getValue());
+    }
+
+    private static HttpClient.Version getHttpVersion(String httpVersion) {
+        if (Constants.HTTP_2.equals(httpVersion)) {
+            return HttpClient.Version.HTTP_2;
+        }
+        return HttpClient.Version.HTTP_1_1;
+    }
+
+    private static SSLContext initSslContext() throws NoSuchAlgorithmException, KeyManagementException {
+        TrustManager[] trustAllCerts = new TrustManager[]{
+                new X509TrustManager() {
+                    public X509Certificate[] getAcceptedIssuers() {
+                        return null;
+                    }
+
+                    public void checkClientTrusted(X509Certificate[] certs, String authType) {
+                    }
+
+                    public void checkServerTrusted(X509Certificate[] certs, String authType) {
+                    }
+                }
+        };
+        SSLContext sslContext = SSLContext.getInstance(Constants.TLS);
+        sslContext.init(null, trustAllCerts, new SecureRandom());
+        return sslContext;
+    }
+
+    private static SSLContext initSslContext(BMap<BString, BString> trustStore) throws Exception {
+        String path = trustStore.getStringValue(StringUtils.fromString(Constants.PATH)).getValue();
+        String password = trustStore.getStringValue(StringUtils.fromString(Constants.PASSWORD)).getValue();
+        InputStream is = new FileInputStream(new File(path));
+        char[] passphrase = password.toCharArray();
+        KeyStore ks = KeyStore.getInstance(Constants.PKCS12);
+        ks.load(is, passphrase);
+        KeyManagerFactory kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        kmf.init(ks, passphrase);
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        tmf.init(ks);
+        SSLContext sslContext = SSLContext.getInstance(Constants.TLS);
+        sslContext.init(kmf.getKeyManagers(), tmf.getTrustManagers(), new SecureRandom());
```

Comment:
As we are only configuring the trustore, Can't we pass null for keymangers without creating the KeyManagerFactory and getting the keymanagers from there?

Reply:
Fixed with https://github.com/ballerina-platform/module-ballerina-jwt/pull/64/commits/69b8c4377ee8e2a243f7ca23982a5c511fe138f6